docs/book/src/reference/testdata/watching/controller.go

```diff
@@ -0,0 +1,218 @@
+Therefore we must specify a custom way of watching those objects.
+This watch logic is complex, so we have split it into a separate method.
+*/
+
+func (r *ConfigDeploymentReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	ctrlBuilder := ctrl.NewControllerManagedBy(mgr).
+		For(&appsv1.ConfigDeployment{}).
+		Owns(&kapps.Deployment{})
+
+	var err error
+	if ctrlBuilder, err = r.watchUserConfigMaps(mgr, ctrlBuilder); err != nil {
+		return err
+	}
+
+	return ctrlBuilder.Complete(r)
+}
+
+/*
+There are two necessary steps that will allow us to watch the `ConfigMaps` that are referenced by `ConfigDeployments`.
+Index the reference field and setup a watch.
+*/
+
+func (r *ConfigDeploymentReconciler) watchUserConfigMaps(mgr ctrl.Manager, ctrlBuilder *builder.Builder) (*builder.Builder, error) {
+	/*
+	Determine the path of the field in the ConfigDeployment CRD that we wish to use as the "object reference".
+	This will be used in both the indexing and watching.
+	*/
+	configMapField := ".spec.configMap"
+
+	/*
+	This field must be indexed by the manager, so that we will be able to lookup `ConfigDeployments` by a referenced `ConfigMap` name.
+	This will allow for quickly answer the question:
+	- If ConfigMap _x_ is updated, which ConfigDeployments are affected?
+	 */
+
+	if err := mgr.GetFieldIndexer().IndexField(context.TODO(), &appsv1.ConfigDeployment{}, configMapField, func(rawObj runtime.Object) []string {
+		// Extract the ConfigMap name from the ConfigDeployment Spec, if one is provided
+		configDeployment := rawObj.(*appsv1.ConfigDeployment)
+		if configDeployment.Spec.ConfigMap == "" {
+			return nil
+		}
+		return []string{configDeployment.Spec.ConfigMap}
+	}); err != nil {
+		return ctrlBuilder, err
+	}
+
+
+	/*
+	Now that we can quickly determine which `ConfigDeployments` need to be reconciled when a `ConfigMap` is updated, we can setup a watch for `ConfigMaps`.
+
+	This Watches() is a controller-runtime API that takes:
+	- A Kind (i.e. `ConfigMap`)
+	- A mapping function that converts a `ConfigMap` object to a list of reconcile requests for `ConfigDeployments`
+	- A list of options for watching the `ConfigMaps`
+	  - In our case, we only want the watch to be triggered when the ResourceVersion of the ConfigMap is changed.
+
+	Because we have already created an index on the configMap reference field, this mapping function is quite straight forward.
+	We first need to list out all `ConfigDeployments` that use `ConfigMap` given in the mapping function.
+	This is done by merely submitting a List request using our indexed field as the field selector.
+
+	When the list of `ConfigDeployments` that reference the `ConfigMap` is found,
+	we just need to loop through the list and create a reconcile request for each one.
+	If an error occurs fetching the list, or no `ConfigDeployments` are found, then no reconcile requests will be returned.
+	 */
+
+	return ctrlBuilder.Watches(
```

Comment:
Your goal explain how to watch a resource that is not managed by the controller, am I right? then, I do not think that you need to do this implementation in predicate to list all. See : 

```go
err := c.Watch(&source.Kind{Type: obj}, &handler.EnqueueRequestForOwner{
		IsController: isConttroller,
		OwnerType:    owner,
	})
```

So, you can pass false in the `isConttroller`. Did you try that?

Reply:
So I think it would be useful to add a longer intro into the use case before diving into the code.

Basically the idea here is that the user has a `ConfigMap` which _they_ manage. However they want the Deployment, managed by the controller, to use this ConfigMap and for the Pods to be updated whenever the ConfigMap changes. So the controller is not controlling the ConfigMap.

This is a pretty common use case, especially for resources such as TLS certs in Secrets. You may want to auto-update pods when the TLS Certs have been renewed (i.e. the secret has been updated). Maybe a more concrete use case would be a good idea? My intention was to be as simple as possible, but I might have missed the mark there.

I'm not 100% certain, but I don't think the logic would be the same using:
```go
err := c.Watch(&source.Kind{Type: obj}, &handler.EnqueueRequestForOwner{
		IsController: false,
		OwnerType:    owner,
	})
```

The ConfigDeployment is not the owner of the ConfigMap, just a user of it. If we used this watch, then we would have to add an owner reference to a ConfigMap for every ConfigDeployment that uses it (not necessarily a 1<->1 mapping).

Reply:
And a somewhat off-topic question about your second example.

Is that any different from using the following?
```go
 func (r *MemcachedReconciler) SetupWithManager(mgr ctrl.Manager) error {
	return ctrl.NewControllerManagedBy(mgr).
		For(&cachev1alpha1.Memcached{}).
		Owns(&corev1.ConfigMap{}).
		Owns(&appsv1.Deployment{}).
		Complete(r)
}
```

Reply:
See [here](https://github.com/dev4devs-com/postgresql-operator/blob/master/pkg/controller/backup/controller.go#L61). I watch a resource `v1.Service` which is not managed by the controller and has another owner. 

IMO we need to the doc with examples; 
- the common basic scenario ( watch a secondary resource managed by the controller)
- how to watch a resource which is owned/managed by another controller in the project ( [here](https://github.com/dev4devs-com/postgresql-operator/blob/master/pkg/controller/backup/controller.go#L61))
- then, how to watch a resource that we do not know the owner/who manages and we want to use predicates to filter as for example by labels.

Reply:
Agreed, I think it would be good to start off basic and through each example get more complex.

For the last example I think it would be good to show using a label filter. But then I think the example I gave would be a good extension on top of that. (Many CRDs I have seen have uses cases like this, and it is pretty confusing for new people who want to use it)

That would give us 4 examples, the first one being extremely simple. And the labels one should be pretty quick as well.